src/tokens/blockchain.service.ts

```diff
@@ -88,15 +104,60 @@ export class BlockchainConnectorService {
     });
   }
 
+  // Check if retry condition matches the err that's been hit
+  private matchesRetryCondition(err: any): boolean {
+    return this.retryCondition != '' && err?.toString().match(this.retryCondition) !== null;
```

Comment:
This will crash if `err` is defined but doesn't have a `toString` method.

Reply:
Is it possible for err to not have a toString() method? I have a feeling all Javascript objects have it, and as `matchesRetryCondition()` is only called from a catch block I think `err` must always have that method? I could add a `typeof err.toString === 'function'` check to make absolutely sure?

Reply:
Technically you can `throw` anything you want, so there's no guarantees. Well-behaved code should only throw things that inherit from `Error` (which implements `toString`), but I'm not sure we should rely on it... An alternative would be using `${err}`, which will try a variety of methods (including `toString`) to coerce the object to a string.

Reply:
Yeah that does feel safer. Updated in the final commit.